crates/parcel_filesystem/src/lib.rs

```diff
@@ -43,6 +43,34 @@ pub trait FileSystem {
   ) -> Result<PathBuf> {
     self.canonicalize_base(path)
   }
+  fn find_ancestor_file<P: AsRef<Path>>(
```

Comment:
I reckon this should be a function that receives filesystem, instead of a trait function.

So 
```rust
  fn find_ancestor_file(
    fs: &impl FileSystem,
    filenames: Vec<String>,
    from: impl AsRef<Path>,
    root: impl AsRef<Path>,
  ) -> Option<PathBuf> {
    for dir in from.as_ref().ancestors() {
      // Break if we hit a node_modules directory
      if let Some(filename) = dir.file_name() {
        if filename == "node_modules" {
          break;
        }
      }

      for name in &filenames {
        let fullpath = dir.join(name);
        if fs.is_file(&fullpath) {
          return Some(fullpath);
        }
      }

      if dir == root.as_ref() {
        break;
      }
    }

    None
  }
```

Reply:
Which crate would you have this fn live in then? Feels like it should be outside of filesystem if it takes it in as a a param imo

Reply:
I'd leave this where you're consuming it if it's not shared yet.

Not sure why it breaks on node_modules

Reply:
Ok, it's only used in package_manager and config within the JS code atm

Reply:
If it's used on two other crates you may want to just export from this one